### PR TITLE
Cellar.sol v2.5 support

### PR DESCRIPTION
**Pull request by cbrit**

### CODE REVIEW

File: docs/api/cellar_v2.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ Represents a call to adaptor an. The cellar must be authorized to call the targe
 | morpho_aave_v3_a_token_collateral_v1_calls | [MorphoAaveV3ATokenCollateralAdaptorV1Calls](#steward-v3-MorphoAaveV3ATokenCollateralAdaptorV1Calls) |  | Represents function calls to the MorphoAaveV3ATokenCollateral V1 |
 | morpho_aave_v3_a_token_p2p_v1_calls | [MorphoAaveV3ATokenP2PAdaptorV1Calls](#steward-v3-MorphoAaveV3ATokenP2PAdaptorV1Calls) |  | Represents function calls to the MorphoAaveV3ATokenP2P V1 |
 | morpho_aave_v3_debt_token_v1_calls | [MorphoAaveV3DebtTokenAdaptorV1Calls](#steward-v3-MorphoAaveV3DebtTokenAdaptorV1Calls) |  | Represents function calls to the MorphoAaveV3DebtToken V1 |
+| balancer_pool_v1_calls | [BalancerPoolAdaptorV1Calls](#steward-v3-BalancerPoolAdaptorV1Calls) |  | Represents function calls to the BalancerPoolAdaptor V1 |
```
Comment (EricBolten) [inline docs/api/cellar_v2.md]: What's going on with this diff into 3.x-main? We already merged the Balancer PR, I'm confused as to what's happening on Github. You've also got a few merge conflicts -- may need to rebase this PR off of 3.x-main.

File: steward/src/cellars/cellar_v2_5.rs [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetShareLockPeriodCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetShareLockPeriodCall {
+                new_lock: string_to_u256(params.new_lock)?,
+            };
+
+            Ok(CellarWithShareLockPeriodV1Calls::SetShareLockPeriod(call).encode())
```
Comment (EricBolten) [inline steward/src/cellars/cellar_v2_5.rs:138]: Just to confirm, this existing function moved from the base cellar to the additional ABI we've added, which is one of the new permutations?

File: steward/src/cellars/cellar_v2_5.rs [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+                new_share_supply_cap: string_to_u256(params.new_cap)?,
+            };
+
+            Ok(CellarV2_5Calls::DecreaseShareSupplyCap(call).encode())
+        }
+    }
+}
+
+/// Encodes calls to the Adaptor contracts
+fn get_encoded_adaptor_calls(data: Vec<AdaptorCall>) -> Result<Vec<AbiAdaptorCall>, Error> {
```
Comment (EricBolten) [inline steward/src/cellars/cellar_v2_5.rs:267]: Is there any difference between this for 2.2 and 2.5? Could we in theory just use a common function for all v2 cellar types when encoding an adaptor call?

**Review by EricBolten — COMMENTED** [review]
A few comments.

File: steward/src/cellars/cellar_v2_5.rs (hunk 2 above)
Comment (cbrit) [reply steward/src/cellars/cellar_v2_5.rs:138]: Yes
https://github.com/PeggyJV/cellar-contracts/blob/main/src/base/permutations/CellarWithShareLockPeriod.sol

File: steward/src/cellars/cellar_v2_5.rs (hunk 3 above)
Comment (cbrit) [reply steward/src/cellars/cellar_v2_5.rs:267]: each `AbiAdaptorCall` (return type) is uniquely scoped.

**Review by EricBolten — APPROVED** [review]
LGTM

**Pull request merged**